tilings/assumptions.py

```diff
@@ -64,24 +20,33 @@ def avoiding(self, obstructions: Iterable[GriddedPerm]) -> "TrackingAssumption":
         obstructions = tuple(obstructions)
         return self.__class__(tuple(gp for gp in self.gps if gp.avoids(*obstructions)))
 
-    @staticmethod
-    def contradictory(tiling: "Tiling") -> bool:
-        return False
-
     def to_jsonable(self) -> dict:
-        d: dict = super().to_jsonable()
-        d["gps"] = [gp.to_jsonable() for gp in self.gps]
-        return d
+        return {"gps": [gp.to_jsonable() for gp in self.gps]}
 
     @classmethod
     def from_dict(cls, d: dict) -> "TrackingAssumption":
         gps = [GriddedPerm.from_dict(gp) for gp in d["gps"]]
         return cls(gps)
 
-    def __lt__(self, other):
-        if isinstance(other, self.__class__):
-            return self.gps < other.gps
-        return super().__lt__(other)
+    def __eq__(self, other) -> bool:
+        try:
```

Comment:
Thie `__eq__` implementations do not follow the python standard for implementation of binary special method. It is recommended to return the `NotImplemented` constant when the operation is not implemented for a type.

```python
def __eq__(self, other: object) -> bool:
    if not isinstance(other, TrackingAssumption):
        return NotImplemented
    return self.gps == other.gps
```
[https://docs.python.org/3/library/constants.html](https://docs.python.org/3/library/constants.html)

Reply:
I'll fix this